pyinfra/operations/pipx.py

```diff
@@ -0,0 +1,100 @@
+"""
+Manage pipx (python) applications.
+"""
+
+from pyinfra import host
+from pyinfra.api import OperationError, operation
+from pyinfra.facts.pipx import PipxEnvironment, PipxPackages
+from pyinfra.facts.server import Path
+
+from .util.packaging import ensure_packages
+
+
+@operation
+def packages(
+    packages=None,
+    present=True,
+    latest=False,
+    requirements=None,
+    pipx="pipx",
+    extra_install_args=None,
+):
+    """
+    Install/remove/update pipx packages.
+
+    + packages: list of packages to ensure
+    + present: whether the packages should be installed
+    + latest: whether to upgrade packages without a specified version
+    + pipx: name or path of the pipx binary to use
+    + extra_install_args: additional arguments to the pipx install command
+
+    Versions:
+        Package versions can be pinned like pip: ``<pkg>==<version>``.
+
+    **Example:**
+
+    .. code:: python
+
+        pipx.packages(
+            name="Install ",
+            packages=["pyinfra"],
+        )
+    """
+
+    # We should always use the --force flag because
+    # if install it's called, that meen the version missmatch
+    # so we need the --force flag to get the good version installed
+    install_command = [pipx, "install", "--force"]
+    if extra_install_args:
+        install_command.append(extra_install_args)
+    install_command = " ".join(install_command)
+
+    uninstall_command = " ".join([pipx, "uninstall"])
+    upgrade_command = " ".join([pipx, "upgrade"])
+
+    # Handle passed in packages
+    if packages:
+        current_packages = host.get_fact(PipxPackages, pipx=pipx)
+
+        if current_packages is None:
```

Comment:
This should never be the case, facts return a default value so can drop this check.